perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java

```diff
@@ -839,17 +839,28 @@ public Object mapRow(ResultSet rs, int i) throws SQLException {
 
 	@Override
 	public List<Attribute> getAttributes(PerunSession sess, Facility facility) throws InternalErrorException {
+		List<Attribute> attributesCache = null;
+
 		if(!CacheManager.isCacheDisabled() && !perun.getCacheManager().wasCacheUpdatedInTransaction()) {
 			List<Attribute> attrs = perun.getCacheManager().getAllNonEmptyAttributes(new Holder(facility.getId(), Holder.HolderType.FACILITY));
-			return this.setValuesOfAttributes(sess, attrs, facility, null);
+			attributesCache = this.setValuesOfAttributes(sess, attrs, facility, null);
 		}
 
 		try {
-			return jdbc.query("select " + getAttributeMappingSelectQuery("fac") + " from attr_names " +
+			List<Attribute> attributesDB = jdbc.query("select " + getAttributeMappingSelectQuery("fac") + " from attr_names " +
 							"left join facility_attr_values fac    on id=fac.attr_id and fac.facility_id=? " +
 							"where namespace=? or (namespace in (?,?) and (attr_value is not null or attr_value_text is not null))",
 					new SingleBeanAttributeRowMapper<>(sess, this, facility), facility.getId(), AttributesManager.NS_FACILITY_ATTR_CORE, AttributesManager.NS_FACILITY_ATTR_DEF, AttributesManager.NS_FACILITY_ATTR_OPT);
 
+			if (attributesCache != null) {
```

Comment:
I would like to see this as some private method. It is repeated too many times. You can always initialize ``attributesCache`` to empty list, since as far as I looked, cache methods return empty list if nothing is found and not null. You can also generally use ``Objects.equals(a,b)`` and don't care about null in any of the variables.